server.js

```diff
@@ -1633,7 +1633,100 @@ app.get('/adminAnalytics', function(req,res) {
     });
 });
 
+app.post('/submitFeedback', function(req, res){
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    if (req.session.user.type === common.userTypes.ADMIN) {
+        return res.status(403).send(common.getError(1002));
+    }
+
+    logger.log(common.formatString('Feedback from {0} regarding {1}', [req.session.user._id, req.body.subject]));
+
+    users.addFeedback(req.session.user._id, req.body.subject, req.body.message, function(err, result) {
+        if (err) {
+            logger.error(err);
+            return res.status(500).send(common.getError(8000));
+        }
+
+        return res.status(201).send('User feedback submitted');
+    });
+});
+
+app.get('/feedback', function(req, res){
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    if (req.session.user.type !== common.userTypes.ADMIN) {
+        return res.status(403).send(common.getError(1002));
+    }
+
+    logger.log(common.formatString('Getting feedback for {0}', [req.session.user.username]));
+
+    users.getFeedback(function(err, result) {
+        if (err) {
+            logger.error(err);
+            return res.status(500).render('feedback-view', {
+                content: null,
+                user: req.session.user
+            });
+        }
+
+        users.getUsersList((err, userObj) => {
+            if (err) {
+                logger.error(err);
+                return res.status(500).render('feedback-view', {
+                    content: [],
+                    user: req.session.user
+                });
+            }
+
+            var usersList = {};
+            for (i in userObj) {
+                var user = userObj[i];
+                usersList[user._id] = [user.fname + ' ' + user.lname, user.username];
+            }
+
+            var data = [];
+            for(i in result) {
+                var tempData = {};
+
+                logger.log("res - " + result[i].uuid);
+
+                logger.log("res2 - " + usersList[result[i].uuid][0]);
+
+                tempData.fullname = usersList[result[i].uuid][0];
+                tempData.username = usersList[result[i].uuid][1];
+                tempData.subject = result[i].subject;
+                tempData.message = result[i].message;
+                tempData.time = result[i].time;
+
+                data.push(tempData);
+            }
+
+            data = data.length === 0 ? null : data;
+
+            return res.status(201).render('feedback-view', {
+                content: data,
+                user: req.session.user
+            });
+        });
+    })
+});
+
+app.post('/removeAllFeedback', function(req, res) {
```

Comment:
comment

Reply:
wait, why do we need to remove feedback?

Reply:
it's an option that the user can do if they want to clear all the feedback

Reply:
I had that function for datagen, so ramy suggested to allow the instructor to use it too.